### DISCUSSION TITLE
Maximum rows: Automatically deleting oldest rows?

### DISCUSSION THREAD

**Discussion by oguruma1218** [supabase/supabase#14374] [Questions]
I'd like to limit the rows of a table to 250,000. After 250,000, I'd like to delete the oldest row to keep the size of the database down.

 This is for a basic logging app that will log the states of various devices.

Can Supabase do this natively?

**Comment by GaryAustin1** [comment]
Supabase is Postgres for the table operations so you can google solutions on postgres and limit table size.
I believe you will find it is not performant to count the table rows and do this on each insert though.

You might be better off picking a max time frame and deleting rows older than that on either an insert trigger function, or with a cron task.

**Comment by oguruma1218** [comment]
After doing a bit of reading, Cron Task seems like it might be a good way to do it… I suppose I could also copy/dump those rows to some other storage outside the database for archival before they’re deleted.
